config/main.py

```diff
@@ -33,6 +35,64 @@ def run_command(command, display_cmd=False, ignore_error=False):
     if proc.returncode != 0 and not ignore_error:
         sys.exit(proc.returncode)
 
+def convert_interface_name(interface_name):
+    """Return alias interface name if default interface is given as argument.
+       Return default interface name if alias name is given as argument.
+    """
+    cmd = 'sonic-cfggen -d --var-json "PORT"'
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+
+    port_dict = json.loads(p.stdout.read())
+
+    if interface_name is not None:
+        for port_name in natsorted(port_dict.keys()):
+            if interface_name == port_name:
+                return port_dict[port_name]['alias']
+            elif interface_name == port_dict[port_name]['alias']:
+                return port_name
+        print "Invalid interface {}".format(interface_name)
+
+def set_interface_mode(mode):
+    tmp = "/tmp/"
+    output = subprocess.check_output(['tty']).strip('\n')
```

Comment:
Agreed. Initially we approched the issue in the above context. Since python is the child process of parent bash (as like for any sessios) we can inherit the status of enviroment variable say SONIC_CLI_ALIAS_MODE to python but we can't revert back the state to TRUE/FALSE back to bash environment variable. .  The change do exists until the script terminates. SONIC_CLI_ALIAS_MODE value cannot be exported to parent bash.

Reply:
Ah, yes ... you can't modify the environment of the parent shell. That's a problem. If we wanted to proceed with this approach, I guess we would need to prompt the user to log out and log back in again for the change to take effect; I think that's quite an inconvenience.

@lguohan: Do you have any alternative suggestions? Otherwise, I believe the tty-based approach may be our best bet.